pg_search/tests/pg_regress/sql/issue-2533.sql

```diff
@@ -0,0 +1,293 @@
+DROP TABLE IF EXISTS users;
+DROP TABLE IF EXISTS products;
+DROP TABLE IF EXISTS orders;
+
+CREATE TABLE users
+(
+    id    serial8 not null primary key,
+    name  text,
+    color varchar,
+    age   varchar
+);
+
+CREATE TABLE products
+(
+    id    serial8 not null primary key,
+    name  text,
+    color varchar,
+    age   varchar
+);
+
+CREATE TABLE orders
+(
+    id    serial8 not null primary key,
+    name  text,
+    color varchar,
+    age   varchar
+);
+
+
+CREATE INDEX idxusers ON users USING bm25 (id, name, color, age)
+    WITH (
+    key_field = 'id',
+    text_fields = '
+            {
+                "name": { "tokenizer": { "type": "keyword" } },
+                "color": { "tokenizer": { "type": "keyword" } },
+                "age": { "tokenizer": { "type": "keyword" } }
+            }'
+    );
+CREATE INDEX idxproducts ON products USING bm25 (id, name, color, age)
+    WITH (
+    key_field = 'id',
+    text_fields = '
+            {
+                "name": { "tokenizer": { "type": "keyword" } },
+                "color": { "tokenizer": { "type": "keyword" } },
+                "age": { "tokenizer": { "type": "keyword" } }
+            }'
+    );
+CREATE INDEX idxorders ON orders USING bm25 (id, name, color, age)
+    WITH (
+    key_field = 'id',
+    text_fields = '
+            {
+                "name": { "tokenizer": { "type": "keyword" } },
+                "color": { "tokenizer": { "type": "keyword" } },
+                "age": { "tokenizer": { "type": "keyword" } }
+            }'
+    );
+
```

Comment:
So: I have to be honest.

This type of test is inscrutable to me, and I think that that is a critical part of the minimization that something like `proptest` would do. How many of these indexes are actually necessary to repro? How many of the inserts? Etc.

I'd like to spend some time seeing whether we can convert #2540 into a proptest, and if so, what it emits for this case.

Reply:
Here is what #2540 looks like as a proptest: https://github.com/paradedb/paradedb/commit/fff6bec7f0d2c873e6e0a17a5b7c90e1c9fdda0d

It reproduces one failure that minimizes to:
```sql
SELECT COUNT(*) FROM orders JOIN products ON orders.color = products.color  WHERE (NOT (orders.name @@@ 'bob')) OR (products.name @@@ 'bob')
SELECT COUNT(*) FROM orders JOIN products ON orders.color = products.color  WHERE (NOT (orders.name  =  'bob')) OR (products.name  =  'bob')
```

Reply:
Opened https://github.com/paradedb/paradedb/pull/2547 for this.